tesseract-core/src/lib.rs

```diff
@@ -205,7 +205,94 @@ impl Schema {
         Ok((sql, header))
     }
 
-    /// Convert user parameters into required default member cuts based on cube defintion.
+    /// Generates SQL to resolve a members locale query.
+    /// Supports resolving multiple locales at the same time.
+    pub fn members_locale_sql(
+        &self,
+        cube_name: &str,
+        level_name: &LevelName,
+        locale: &str
+    ) -> Result<(String, Vec<String>), Error> // Sql and then Header
+    {
+        let locales: Vec<String> = locale.split(",").map(|s| s.to_string()).collect();
+
+        let cube = self.cubes.iter()
+            .find(|cube| &cube.name == &cube_name)
+            .ok_or(format_err!("Could not find cube"))?;
+
+        let dim = cube.dimensions.iter()
+            .find(|dim| dim.name == level_name.dimension)
+            .ok_or(format_err!("could not find dimension for level name"))?;
+        let hier = dim.hierarchies.iter()
+            .find(|hier| hier.name == level_name.hierarchy)
+            .ok_or(format_err!("could not find hierarchy for level name"))?;
+        let level = hier.levels.iter()
+            .find(|lvl| lvl.name == level_name.level)
+            .ok_or(format_err!("could not find level for level name"))?;
+
+        let table = hier.table.clone().unwrap_or_else(|| cube.table.clone());
+
+        let key_column = level.key_column.clone();
+        let mut header = vec!["ID".into()];
+        let mut name_columns: Vec<String> = vec![];
+
+        // TODO: have a check that there can't be inline table and regular table at the same time.
```

Comment:
As as note, I think I have somewhere else in the code a TODO to make table an enum instead of having different fields for in-line and regular tables.